### PR TITLE
Refactor "organizations" to "teams"

### PR DESCRIPTION
**Pull request by ZmagoD**
https://biosistemika.atlassian.net/browse/SCI-939

Check also the submodules

### CODE REVIEW

File: app/models/user_team.rb [hunk 1]
```diff
@@ -1,45 +1,45 @@
...
   belongs_to :assigned_by, foreign_key: 'assigned_by_id', class_name: 'User'
-  belongs_to :organization, inverse_of: :user_organizations
+  belongs_to :team, inverse_of: :user_teams
 
   before_destroy :destroy_associations
   after_create :create_samples_table_state
 
   def role_str
-    I18n.t("user_organizations.enums.role.#{role.to_s}")
+    I18n.t("user_teams.enums.role.#{role.to_s}")
```
Comment (Ducz0r) [inline app/models/user_team.rb]: Hound is love, Hound is life :dog:

File: app/models/team.rb [hunk 2]
```diff
@@ -128,26 +128,26 @@ def import_samples(sheet, mappings, user)
...
               sample.create_sample_type(
                 name: value,
-                organization_id: id
+                team_id: id
               )
             end
             sample.save
           elsif index == sgroup_index
-            sgroup = SampleGroup.where(name: value, organization_id: id).take();
+            sgroup = SampleGroup.where(name: value, team_id: id).take();
```
Comment (Ducz0r) [inline app/models/team.rb]: Hound is love, Hound is life :dog:

File: app/models/team.rb [hunk 3]
```diff
@@ -128,26 +128,26 @@ def import_samples(sheet, mappings, user)
           # We need to have sample saved before messing with custom fields (they
           # need sample id)
           if index == stype_index
-            stype = SampleType.where(name: value, organization_id: id).take();
+            stype = SampleType.where(name: value, team_id: id).take();
```
Comment (Ducz0r) [inline app/models/team.rb]: Hound is love, Hound is life :dog:

File: app/datatables/sample_datatable.rb [hunk 4]
```diff
@@ -294,9 +294,9 @@ def sort_records(records)
             .select('"samples"."id"')
             .distinct
 
-          # grabs the ids that are not the previous one but are still of the same organization
+          # grabs the ids that are not the previous one but are still of the same team
           unassigned = Sample
-            .where('"samples"."organization_id" = ?', @organization.id)
+            .where('"samples"."team_id" = ?', @team.id)
```
Comment (Ducz0r) [inline app/datatables/sample_datatable.rb]: Hound is love, Hound is life :dog:

Probably better to assign Sample into a new line?

File: app/datatables/sample_datatable.rb [hunk 5]
```diff
@@ -259,9 +259,9 @@ def sort_records(records)
...
```
Comment (Ducz0r) [inline app/datatables/sample_datatable.rb]: Hound is love, Hound is life :dog:

Probably better to move Sample into new line?

File: app/controllers/users/settings_controller.rb [hunk 6]
```diff
@@ -367,20 +367,20 @@ def load_user
...
-    # Don't allow the user to modify UserOrganization-s if he's not admin,
-    # unless he/she is modifying his/her UserOrganization
-    if current_user != @user_org.user and
-      !is_admin_of_organization(@user_org.organization)
+  def check_user_team_permission
+    @user_team = UserTeam.find_by_id(params[:user_team_id])
+    @team = @user_team.team
+    # Don't allow the user to modify UserTeam-s if he's not admin,
+    # unless he/she is modifying his/her UserTeam
+    if current_user != @user_team.user and
```
Comment (Ducz0r) [inline app/controllers/users/settings_controller.rb]: Hound is love, Hound is life :dog:

File: app/controllers/users/settings_controller.rb [hunk 7]
```diff
@@ -271,22 +271,22 @@ def destroy_user_organization
...
       @user
-      .user_organizations
-      .includes(organization: :users)
+      .user_teams
+      .includes(team: :users)
       .where(role: 1..2)
       .order(created_at: :asc)
-      .map { |uo| uo.organization }
-    @member_of = @orgs.count
+      .map { |uo| uo.team }
```
Comment (Ducz0r) [inline app/controllers/users/settings_controller.rb]: Hound is love, Hound is life :dog:

File: app/controllers/users/settings_controller.rb [hunk 8]
```diff
@@ -254,14 +254,14 @@ def destroy_user_organization
       if !invalid
         if params[:leave] then
           flash[:notice] = I18n.t(
-            "users.settings.organizations.index.leave_flash",
-            org: @user_org.organization.name
+            "users.settings.teams.index.leave_flash",
```
Comment (Ducz0r) [inline app/controllers/users/settings_controller.rb]: Hound is love, Hound is life :dog:

File: app/controllers/users/settings_controller.rb [hunk 9]
```diff
@@ -53,198 +53,198 @@ def update_preferences
...
-            partial: "users/settings/organizations/destroy_user_organization_modal_body.html.erb",
-            locals: { user_organization: @user_org }
+            partial: "users/settings/teams/destroy_user_team_modal_body.html.erb",
+            locals: { user_team: @user_team }
           }),
           heading: I18n.t(
-            "users.settings.organizations.edit.destroy_uo_heading",
-            user: escape_input(@user_org.user.full_name),
-            org: escape_input(@user_org.organization.name)
+            "users.settings.teams.edit.destroy_uo_heading",
```
Comment (Ducz0r) [inline app/controllers/users/settings_controller.rb]: Hound is love, Hound is life :dog:

File: app/controllers/users/settings_controller.rb [hunk 10]
```diff
@@ -53,198 +53,198 @@ def update_preferences
...
           html: render_to_string({
-            partial: "users/settings/organizations/leave_user_organization_modal_body.html.erb",
-            locals: { user_organization: @user_org }
+            partial: "users/settings/teams/leave_user_team_modal_body.html.erb",
+            locals: { user_team: @user_team }
           }),
           heading: I18n.t(
-            "users.settings.organizations.index.leave_uo_heading",
-            org: escape_input(@user_org.organization.name)
+            "users.settings.teams.index.leave_uo_heading",
```
Comment (Ducz0r) [inline app/controllers/users/settings_controller.rb]: Hound is love, Hound is life :dog:

File: app/models/protocol.rb [hunk 11]
```diff
@@ -67,10 +67,10 @@ class Protocol < ActiveRecord::Base
...
 
   def self.search(user, include_archived, query = nil, page = 1)
-    org_ids =
-      Organization
-      .joins(:user_organizations)
-      .where("user_organizations.user_id = ?", user.id)
+    team_ids =
+      Team
+      .joins(:user_teams)
+      .where("user_teams.user_id = ?", user.id)
```
Comment (Ducz0r) [inline app/models/protocol.rb]: Hound is love, Hound is life :dog:

File: lib/tasks/db_users.rake [hunk 12]
```diff
@@ -89,36 +89,36 @@ namespace :db do
...
-        org_names.each do |org_name|
-          org = Organization.order(created_at: :desc).where(name: org_name).first
-          if org.blank? then
-            org = Organization.create({ name: org_name[0..99] })
+        # Add/fetch teams if needed
+        team_ids = []
+        team_names.each do |team_name|
+          team = Team.order(created_at: :desc).where(name: team_name).first
+          if team.blank? then
+            team = Team.create({ name: team_name[0..99] })
```
Comment (Ducz0r) [inline lib/tasks/db_users.rake]: Hound is love, Hound is life :dog:

File: lib/tasks/db_users.rake [hunk 13]
```diff
@@ -89,36 +89,36 @@ namespace :db do
...
-      org_names = []
+    puts "Do you want Rails to create default user's team? (T/F)"
+    create_team = $stdin.gets.to_s.strip == "T"
+    puts "Type names of any additional teams you want the user to be admin of (delimited with ','), or leave blank"
+    team_names = $stdin.gets.to_s.strip
+    if team_names.empty?
+      team_names = []
     else
-      org_names = org_names.split(",").collect { |n| n.strip }
+      team_names = team_names.split(",").collect { |n| n.strip }
```
Comment (Ducz0r) [inline lib/tasks/db_users.rake]: Hound is love, Hound is life :dog:

File: lib/tasks/db_users.rake [hunk 14]
```diff
@@ -89,36 +89,36 @@ namespace :db do
...
       password = generate_user_password
     end
-    puts "Do you want Rails to create default user's organization? (T/F)"
-    create_org = $stdin.gets.to_s.strip == "T"
-    puts "Type names of any additional organizations you want the user to be admin of (delimited with ','), or leave blank"
-    org_names = $stdin.gets.to_s.strip
-    if org_names.empty?
-      org_names = []
+    puts "Do you want Rails to create default user's team? (T/F)"
+    create_team = $stdin.gets.to_s.strip == "T"
```
Comment (Ducz0r) [inline lib/tasks/db_users.rake]: Hound is love, Hound is life :dog:

File: lib/tasks/db_users.rake [hunk 15]
```diff
@@ -43,26 +43,26 @@ namespace :db do
...
-            user_orgs
+          team_ids =
+            user_teams
             .split(",")
             .collect{ |o| o.strip }
             .uniq
-            .select{ |o| orgs.include? o }
-            .collect{ |o| orgs[o]["id"] }
+            .select{ |o| teams.include? o }
+            .collect{ |o| teams[o]["id"] }
```
Comment (Ducz0r) [inline lib/tasks/db_users.rake]: Hound is love, Hound is life :dog:

File: lib/tasks/db_users.rake [hunk 16]
```diff
@@ -43,26 +43,26 @@ namespace :db do
...
-          org_ids =
-            user_orgs
+          team_ids =
+            user_teams
             .split(",")
             .collect{ |o| o.strip }
             .uniq
-            .select{ |o| orgs.include? o }
-            .collect{ |o| orgs[o]["id"] }
+            .select{ |o| teams.include? o }
```
Comment (Ducz0r) [inline lib/tasks/db_users.rake]: Hound is love, Hound is life :dog:

File: lib/tasks/db_users.rake [hunk 17]
```diff
@@ -43,26 +43,26 @@ namespace :db do
             password = generate_user_password
           end
 
-          user_orgs = user_hash["organizations"]
-          if user_orgs.blank?
-            user_orgs = ""
+          user_teams = user_hash["teams"]
+          if user_teams.blank?
+            user_teams = ""
```
Comment (Ducz0r) [inline lib/tasks/db_users.rake]: Hound is love, Hound is life :dog:

File: lib/tasks/db_users.rake [hunk 18]
```diff
@@ -43,26 +43,26 @@ namespace :db do
             password = generate_user_password
           end
 
-          user_orgs = user_hash["organizations"]
-          if user_orgs.blank?
-            user_orgs = ""
+          user_teams = user_hash["teams"]
```
Comment (Ducz0r) [inline lib/tasks/db_users.rake]: Hound is love, Hound is life :dog:

File: lib/tasks/db_users.rake [hunk 19]
```diff
@@ -4,35 +4,35 @@ include UsersGenerator
...
+        # Create teams
+        teams.each do |k, team_hash|
+          team = Team.order(created_at: :desc).where(name: team_hash["name"]).first
+          if team.blank?
+            team = Team.create({
+              name: team_hash["name"][0..99]
             })
           end
-          org_hash["id"] = org.id
+          team_hash["id"] = team.id
```
Comment (Ducz0r) [inline lib/tasks/db_users.rake]: Hound is love, Hound is life :dog:

File: lib/tasks/db_users.rake [hunk 20]
```diff
@@ -4,35 +4,35 @@ include UsersGenerator
...
-        orgs.each do |k, org_hash|
-          org = Organization.order(created_at: :desc).where(name: org_hash["name"]).first
-          if org.blank?
-            org = Organization.create({
-              name: org_hash["name"][0..99]
+        # Create teams
+        teams.each do |k, team_hash|
+          team = Team.order(created_at: :desc).where(name: team_hash["name"]).first
+          if team.blank?
+            team = Team.create({
```
Comment (Ducz0r) [inline lib/tasks/db_users.rake]: Hound is love, Hound is life :dog:

File: lib/tasks/db_users.rake [hunk 21]
```diff
@@ -4,35 +4,35 @@ include UsersGenerator
...
 
-        # Create organizations
-        orgs.each do |k, org_hash|
-          org = Organization.order(created_at: :desc).where(name: org_hash["name"]).first
-          if org.blank?
-            org = Organization.create({
-              name: org_hash["name"][0..99]
+        # Create teams
+        teams.each do |k, team_hash|
+          team = Team.order(created_at: :desc).where(name: team_hash["name"]).first
```
Comment (Ducz0r) [inline lib/tasks/db_users.rake]: Hound is love, Hound is life :dog:

File: lib/tasks/db_users.rake [hunk 22]
```diff
@@ -4,35 +4,35 @@ include UsersGenerator
...
         users = yaml.select{ |k, v| /user_[0-9]+/ =~ k }
 
-        # Create organizations
-        orgs.each do |k, org_hash|
-          org = Organization.order(created_at: :desc).where(name: org_hash["name"]).first
-          if org.blank?
-            org = Organization.create({
-              name: org_hash["name"][0..99]
+        # Create teams
+        teams.each do |k, team_hash|
```
Comment (Ducz0r) [inline lib/tasks/db_users.rake]: Hound is love, Hound is life :dog:

File: app/models/sample.rb [hunk 23]
```diff
@@ -24,12 +24,12 @@ def self.search(
...
+    current_team = nil
   )
-    org_ids =
-      Organization
-      .joins(:user_organizations)
-      .where("user_organizations.user_id = ?", user.id)
+    team_ids =
+      Team
+      .joins(:user_teams)
+      .where("user_teams.user_id = ?", user.id)
```
Comment (Ducz0r) [inline app/models/sample.rb]: Hound is love, Hound is life :dog:

File: app/models/protocol.rb [hunk 24]
```diff
@@ -40,24 +40,24 @@ class Protocol < ActiveRecord::Base
...
-    # Private protocol must have unique name inside its organization & user scope
-    protocol.validates_uniqueness_of :name, scope: [:organization, :added_by], conditions: -> { where(protocol_type: Protocol.protocol_types[:in_repository_private]) }
+    # Private protocol must have unique name inside its team & user scope
+    protocol.validates_uniqueness_of :name, scope: [:team, :added_by], conditions: -> { where(protocol_type: Protocol.protocol_types[:in_repository_private]) }
   end
   with_options if: :in_repository_archived? do |protocol|
-    # Archived protocol must have unique name inside its organization & user scope
-    protocol.validates_uniqueness_of :name, scope: [:organization, :added_by], conditions: -> { where(protocol_type: Protocol.protocol_types[:in_repository_archived]) }
+    # Archived protocol must have unique name inside its team & user scope
+    protocol.validates_uniqueness_of :name, scope: [:team, :added_by], conditions: -> { where(protocol_type: Protocol.protocol_types[:in_repository_archived]) }
```
Comment (Ducz0r) [inline app/models/protocol.rb]: Hound is love, Hound is life :dog:

Comment (ZmagoD) [inline app/models/protocol.rb]: hound is a dick

**Comment by Ducz0r** [comment]
I'm greenlighting this PR! Good luck! _A day may come, when the courage of men fails, when we forsake our friends and break all bonds of Fellowship, but it is not this day! This day we merge!_

**Pull request merged**